src/pages/MainPage.tsx

```diff
@@ -186,69 +189,83 @@ function MainPage() {
   };
 
   // Only want this to trigger once to grab user token and id.
-  useEffect(async () => {
-    socket.disconnect();
+  useEffect(() => {
+    void (async () => {
+      socket.disconnect();
 
-    const token = localStorage.getItem('token');
-    const id = localStorage.getItem('id');
+      const token = localStorage.getItem('token');
+      const id = localStorage.getItem('id');
 
-    const userInfo = await getUser(token, id);
+      if (!token || !id) {
+        history.replace('/landing');
+        return;
+      }
 
-    if (!userInfo) {
-      return;
-    }
+      const userInfo = await getUser(token ?? '', id ?? '');
```

Comment:
Nitpick: The `??` isn't necessary here because of the check above (same applies below)